toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1085,6 +1092,15 @@ func (g *PkgGraph) AddGoalNode(goalName string, packages []*pkgjson.PackageVer,
 		}
 	}
 
+	//Now grab the nodes by SRPM path if we set useSRPMPath
+	for _, node := range g.AllRunNodes() {
+		if srpmPathSet[node.SrpmPath] {
+			logger.Log.Tracef("Found %s to satisfy SRPM linked goal %s", node, node.SrpmPath)
+			goalEdge := g.NewEdge(goalNode, node)
+			g.SetEdge(goalEdge)
+		}
+	}
```

Comment:
Let's save some time on `g.AllRunNodes()`, if we don't need it:
```suggestion
	if useSRPMPath {
		for _, node := range g.AllRunNodes() {
			if srpmPathSet[node.SrpmPath] {
				logger.Log.Tracef("Found %s to satisfy SRPM linked goal %s", node, node.SrpmPath)
				goalEdge := g.NewEdge(goalNode, node)
				g.SetEdge(goalEdge)
			}
		}
	}
```

Reply:
Ah, good call. I vaguely remember thinking that would make sense a while ago and yet here we are...